### DISCUSSION TITLE
Allow Spacing Between Commands

### DISCUSSION THREAD

**Discussion by jackHedaya** [mtxr/vscode-sqltools#887] [Q&A] [answered]
How can I add spacing between commands such as 

```
CREATE EXTENSION IF NOT EXISTS "uuid-ossp";

CREATE TABLE posts (
  id uuid DEFAULT uuid_generate_v4(),
  title VARCHAR NOT NULL,
  body TEXT NOT NULL,
  published BOOLEAN NOT NULL DEFAULT 'f'
)
```

It auto formats to:
```
CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
CREATE TABLE posts (
  id uuid DEFAULT uuid_generate_v4(),
  title VARCHAR NOT NULL,
  body TEXT NOT NULL,
  published BOOLEAN NOT NULL DEFAULT 'f'
)
```

**Comment by gimbo** [comment, marked as answer]
Look for the `sqltools.format` setting; e.g. in my `settings.json` I have:

```
  "sqltools.format": {
    "language": "sql",
    "linesBetweenQueries": 3,
    "reservedWordCase": "upper"
  },
```

That actually creates *two* blank lines between queries — I guess the setting is really "the number of linebreaks" so the default would be 1 which just puts in a single linebreak, giving the output you see.

If you want a single blank line between queries, you want `"linesBetweenQueries": 2`

**Comment by hitesh-pathak** [comment]
Is there a way to allow blank lines within the same query, I have some long queries, and a blank line helps a lot with readability. But using the formatter, the blank lines are completely removed...

They are in the same query so the above configuration does not work.
